lib/ClimaCoreTempestRemap/src/onlineremap.jl

```diff
@@ -68,18 +73,73 @@ function remap!(
     return target
 end
 
-"""
-    remap!(target::Field, R::LinearMap, source::Field)
-
-Applies the remapping `R` to a `source` Field, storing the result in `target`.
-"""
+# This version of this function is used for distributed remapping
 function remap!(target::Fields.Field, R::LinearMap, source::Fields.Field)
-    @assert axes(source) == R.source_space
-    @assert axes(target) == R.target_space
-    # we use the tempestremap cgll representation
-    # it will set the redundant nodes to zero
-    remap!(Fields.field_values(target), R, Fields.field_values(source))
-    return target
+    if Spaces.topology(axes(target)).context isa
+       ClimaComms.SingletonCommsContext
+        @assert axes(source) == R.source_space
+        @assert axes(target) == R.target_space
+        # we use the tempestremap cgll representation
+        # it will set the redundant nodes to zero
+        remap!(Fields.field_values(target), R, Fields.field_values(source))
+        return target
+    else
+        # For now, the source data must be on a non-distributed space
+        @assert Spaces.topology(axes(source)).context isa
+                ClimaComms.SingletonCommsContext
+
+        target_array = parent(target)
+        source_array = parent(source)
+
+        fill!(target_array, zero(eltype(target_array)))
+        Nf = size(target_array, 3)
+
+        # ideally we would use the tempestremap dgll (redundant node) representation
+        # unfortunately, this doesn't appear to work quite as well (for out_type = dgll) as the cgll
+        for (n, wt) in enumerate(R.weights)
+            # choose all global source indices
+            #  (for simple distr. remapping with broadcasted source data, no halo exchange)
+            is, js, es = map(
+                x -> x[1],
+                (
+                    view(R.source_idxs[1], n),
+                    view(R.source_idxs[2], n),
+                    view(R.source_idxs[3], n),
+                ),
+            )
+
+            # choose only the target inds local to this process (skip inds from other processes)
+            t_gidx3 = view(R.target_idxs[3], n)[1]
```

Comment:
I wonder if we could be a bit more verbose here. Maybe `target_elem_gidx` or something like that?